build_modules/lib/src/common.dart

```diff
@@ -34,18 +34,14 @@ Future<File> createPackagesFile(Iterable<AssetId> allAssets) async {
 enum ModuleStrategy { fine, coarse }
 
 ModuleStrategy moduleStrategy(BuilderOptions options) {
-  if (options.isRoot) {
-    var config = options.config['strategy'] as String ?? 'coarse';
-    switch (config) {
-      case 'coarse':
-        return ModuleStrategy.coarse;
-      case 'fine':
-        return ModuleStrategy.fine;
-      default:
-        throw 'Unexpected ModuleBuilder strategy: $config';
-    }
-  } else {
-    return ModuleStrategy.coarse;
+  var config = options.config['strategy'] as String ?? 'coarse';
```

Comment:
Why the removal of the root conditional? Won't that slow things down?

Reply:
Previously injecting a fine strategy module in a dependency somewhere would cause all packages that depend on that package to also get the fine strategy, which I believe is why we silently ignore the config before.

Now you can mix and match the strategies without that cascading behavior - so you don't get the same behavior. Only the specific packages that opted into the fine strategy would get it.

That does mean that users adding the fine strategy but really only intending it to be an option for local development will end up with the fine strategy when people depend on them though... so you would end up needing a second build.yaml or something for local dev. Hopefully that is a rare case, but its hard to really say what the right call is here, silently ignoring it doesn't seem right any more though.